search-service/src/main/java/de/cxp/ocs/elasticsearch/query/builder/DisMaxQueryFactory.java

```diff
@@ -0,0 +1,75 @@
+package de.cxp.ocs.elasticsearch.query.builder;
+
+import de.cxp.ocs.config.Field;
+import de.cxp.ocs.config.FieldConfigAccess;
+import de.cxp.ocs.config.QueryBuildingSetting;
+import de.cxp.ocs.elasticsearch.model.query.ExtendedQuery;
+import de.cxp.ocs.elasticsearch.model.util.EscapeUtil;
+import de.cxp.ocs.elasticsearch.query.TextMatchQuery;
+import de.cxp.ocs.spi.search.ESQueryFactory;
+import lombok.Getter;
+import lombok.Setter;
+import org.elasticsearch.index.query.*;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static de.cxp.ocs.config.QueryBuildingSetting.acceptNoResult;
+import static de.cxp.ocs.util.ESQueryUtils.validateSearchFields;
+
+public class DisMaxQueryFactory implements ESQueryFactory {
+    private final Map<QueryBuildingSetting, String> queryBuildingSettings = new HashMap<>();
+    private final Map<String, Float> masterFields = new HashMap<>();
+    private final Map<String, Float> variantFields = new HashMap<>();
+    @Getter
+    @Setter
+    private String name;
+
+    @Override
+    public void initialize(String name, Map<QueryBuildingSetting, String> settings, Map<String, Float> fieldWeights, FieldConfigAccess fieldConfig) {
+        if (name != null) this.name = name;
```

Comment:
This check is not necessary, as the field name is per default null, just remove the if condition.